src/lock/runner_service.go

```diff
@@ -0,0 +1,76 @@
+package lock
+
+import (
+	"sync"
+
+	"github.com/divideandconquer/go-consul-client/src/config"
+	"github.com/promoboxx/go-discovery/src/discovery"
+)
+
+// RunnerServer is the interface to allow operations on a RunnerServer
+type RunnerServer interface {
+	Run() error
+	Stop() *sync.WaitGroup
+}
+
+type runner struct {
+	runner *Runner
+	state  string
+}
+
+func (r *runner) run() error {
+	err := r.runner.Run()
+	if err != nil {
+		return err
+	}
+	r.state = "running"
+	return nil
+}
+
+func (r *runner) stop() *sync.WaitGroup {
+	if r.state == "running" {
+		return r.runner.Stop()
+	}
+	return nil
+}
+
+type runnerServer struct {
+	environment string
+	serviceName string
+	conf        config.Loader
+	finder      discovery.Finder
+	tracer      Tracer
+	runners     []*runner
+	// client manager field here?
+}
+
+// NewRunnerServer returns a RunnerServer
+func NewRunnerServer(env, serviceName string, conf config.Loader, finder discovery.Finder, tracer Tracer) RunnerServer {
+	ret := &runnerServer{environment: env, serviceName: serviceName, conf: conf, finder: finder, tracer: tracer}
+	ret.init()
```

Comment:
The init() function is different between facebookad and facebookpage... what should we extract from it? do we even need to extract it out and do we need this NewRunnerServer function? @tsharkey

Reply:
What are the differences between the two functions?